data/lib/core/player.lua

```diff
@@ -539,6 +539,11 @@ function Player.updateClientExpDisplay(self)
 	-- Experience bonus (includes server rates)
 	local expGainRate = 100 * Game.getExperienceStage(self:getLevel())
 	self:setClientExpDisplay(expGainRate)
+
+	-- Stamina bonus
+	local staminaMinutes = self:getStamina()
+	local staminaBonus = (staminaMinutes > 2340) and 150 or ((staminaMinutes < 840) and 50 or 100)
```

Comment:
Extract it to a function

```lua
local function getStaminaBonus(staminaMinutes)
  ...
end
```

And use regular if branching to make it clearer what it does